### PR TITLE
[SPARK-41546][CONNECT][PYTHON] `pyspark_types_to_proto_types` should support StructType.

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, `pyspark_types_to_proto_types` used to transform pyspark datatypes to protobuffer datatypes.
But it only supports the primary data type, no struct type.
Many connect API need to transform pyspark struct type to protobuffer struct type. For example, `createDataFrame`, `DataFrame.to` and so on.

### Why are the changes needed?
This PR let `pyspark_types_to_proto_types` support `StructType`.

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
New tests.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @HyukjinKwon @zhengruifeng @amaliujia

File: python/pyspark/sql/connect/types.py [hunk 1]
```diff
@@ -75,6 +75,15 @@ def pyspark_types_to_proto_types(data_type: DataType) -> pb2.DataType:
...
         ret.day_time_interval.end_field = data_type.endField
+    elif isinstance(data_type, StructType):
+        fields = []
+        for field in data_type.fields:
+            struct_field = pb2.DataType.StructField()
+            struct_field.name = field.name
+            struct_field.data_type.CopyFrom(pyspark_types_to_proto_types(field.dataType))
+            struct_field.nullable = field.nullable
+            fields.append(struct_field)
+        ret.struct.fields.extend(fields)
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/types.py:86]:
```suggestion
             ret.struct.fields.append(struct_field)
```

**Review by grundprinzip — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by beliefer** [comment]
@grundprinzip @HyukjinKwon @zhengruifeng Thank you for all !

**Comment by amaliujia** [comment]
late LGTM!

**Pull request closed without merging**